CodingChallenge/MusicSearch/MusicSearchWorker.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  MusicSearchWorker.swift
+//  CodingChallenge
+//
+//  Created by Edgar Barocio on 6/22/19.
+//  Copyright (c) 2019 Edgar Barocio. All rights reserved.
+//
+//  This file was generated by the Clean Swift Xcode Templates so
+//  you can apply clean architecture to your iOS and Mac projects,
+//  see http://clean-swift.com
+//
+
+import UIKit
+
+class MusicSearchWorker {
+    
+    var songResults:[SongResponse] = []
+    let queryService = QueryService()
+    
+    func performSearch(searchTerm:String, completionHandler: @escaping ([SongResponse]) -> Void) {
+        queryService.getSearchResults(searchTerm: searchTerm) { results , errorMessage in
+            if let results = results {
+                self.songResults = results
+                DispatchQueue.main.async {
+                    completionHandler(self.songResults)
+                }
+            } else {
+                DispatchQueue.main.async {
+                    completionHandler([])
+                }
+            }
+            
+            if !errorMessage.isEmpty {
+                print("Search error: " + errorMessage)
+            }
```

Comment:
Consider passing the error up with your completion handler and showing a quick UIAlertController. For super extra bonus points, look at Result in Alamofire and use Generics for maximum power.

Reply:
Yes, it's overkill, but it shows to employers that you know wtf is up.